web-wallet/src/routes/(app)/dashboard/+page.svelte

```diff
@@ -2,20 +2,51 @@
 
 <script>
   import { filterWith, hasKeyValue } from "lamb";
-  import { mdiContain, mdiDatabaseOutline, mdiSwapVertical } from "@mdi/js";
+  import {
+    mdiContain,
+    mdiDatabaseOutline,
+    mdiSwapVertical,
+    mdiSync,
+  } from "@mdi/js";
   import { Icon } from "$lib/dusk/components";
   import { DashboardNav, Transactions } from "$lib/components";
   import { settingsStore, walletStore } from "$lib/stores";
   import { contractDescriptors } from "$lib/contracts";
 
   const { dashboardTransactionLimit, language } = $settingsStore;
 
+  /**
+   * @param {string} contract
+   */
+  function getIconsForContract(contract) {
+    /**
+     * @type {Array.<DashboardNavItemIconProp>}
+     */
+    let icons = [{ path: "" }];
+
+    switch (contract) {
```

Comment:
Maybe easier with an object as a map and `return iconsMap[contract] ?? [{ path: "" }]`? (if a default case is needed at all)

Reply:
I think we will need to refactor this once the token swap work @deuch13 is working on is complete anyway.  I'm not sure how or if the icon mapping will be sufficient, but I suspect it will need a number of changes.  I suggest we make a more permanent solution at that point.

Reply:
To my eyes that's one more reason to keep it as simple as possible.
If we can restrict this:
```js
/** @param {string} contract */
```
to be a `ContractType` instead (union type), we can also ditch the default case, hence also the function, and the resulting code will have no branches at all.

Anyway it's a minor thing and we can think about it later if you prefer.

Reply:
(I meant the `ContractDescriptor.id` property, not the function's param, as the goal is to ditch the function if possible)

Reply:
I'm talking about the changes needed with Alex now, and he will propose a new structure in his PR for the swap interface.